src/components/Header.tsx

```diff
@@ -0,0 +1,62 @@
+import {
+  memo,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+
+interface Props {
+  addTempTodo: (title: string) => void,
+  disabled: boolean,
+  toggleAllTodos: () => void,
+  allCompleted: boolean,
+}
+
+export const Header: React.FC<Props> = memo(({
+  addTempTodo,
+  disabled,
+  toggleAllTodos,
+  allCompleted,
+}) => {
+  const [title, setTitle] = useState('');
+
+  const inputRef = useRef<HTMLInputElement | null>(null);
+
+  useEffect(() => {
+    inputRef.current?.focus();
+  }, []);
+
+  const handleSubmit = (e: React.FormEvent) => {
+    e.preventDefault();
+    addTempTodo(title);
+    setTitle('');
+  };
+
+  return (
+    <header className="todoapp__header">
+
+      <button
+        aria-label="toggleAll"
+        type="button"
+        className={`todoapp__toggle-all ${allCompleted && 'active'}`}
```

Comment:
it is better to use libraries like `classnames` or `clsx` for conditional classNames